matchzoo/inputs/naive_transformer.py

```diff
@@ -0,0 +1,169 @@
+"""Contains the base Transformer class for  all transformers."""
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+from collections import defaultdict
+import numbers
+import copy
+
+from matchzoo import engine
+
+
+class NaiveTransformer(engine.BaseTransformer):
+    """The simple transformer for MatchZoo.
+
+    Examples:
+        >>> X = [("This is the matchzoo toolkit.", "Here is matchzoo.")]
+        >>> transformer = NaiveTransformer()
+        >>> transformer.params['max_df'] = 0.999
+        >>> transformer.params['min_df'] = 0.001
+        >>> transformer.params['max_vocab_size'] = 30000
+        >>> transformer.params['analyzer'] = 'word'
+        >>> transformer.params['stop_words'] = set(['is', 'are'])
+        >>> transformer.params['vocabulary'] = {'a':0, 'b':1}
+        >>> transformer.params['fixed_vocab'] = False
+        >>> X = transformer.fit_transform(X)
+
+    """
+
+    @classmethod
+    def get_default_params(cls) -> engine.TransformerParams:
+        """:return: model default parameters."""
+        params = engine.TransformerParams()
+
+        params['max_df'] = 1.0
+        params['min_df'] = 0.0
+        params['max_vocab_size'] = 2147483648  # 2**31 - 1
+        params['num_of_docs'] = 1
+        params['fixed_vocab'] = False
+        return params
+
+    def build_vocabulary(self, X, fixed_vocab=False):
+        """Create the vocabulary as well as mapped X.
+
+        If fixed_vocab is False, then generate the vocabulary.
+        """
+        if fixed_vocab:
+            vocabulary = self._params['vocabulary']
+        else:
+            # Add a new value when a new vocabulary item is seen
+            vocabulary = defaultdict()
+            vocabulary.default_factory = vocabulary.__len__
+
+        analyze = self.build_analyzer()
+        new_X = copy.deepcopy(X)
+        for idx, pair in enumerate(X):
+            # id1 = instance['id1']
+            # id2 = instance['id2']
+            pair_feature = ([], [])
+            for tid, text in enumerate(pair):
+                for feature in analyze(text):
+                    try:
+                        feature_idx = vocabulary[feature]
+                        pair_feature[tid].append(feature_idx)
+                    except KeyError:
+                        # Ignore out-of-vocabulary items for fixed_vocab=True
+                        continue
+            new_X[idx] = pair_feature
+
+        if not fixed_vocab:
+            # disable defaultdic behaviour
+            vocabulary = dict(vocabulary)
+            if not vocabulary:
+                raise ValueError("empty vocabulary; perhaps the documents only"
+                                 " contain stop words.")
+        return vocabulary, new_X
+
+    def _limit_vocabulary(self, X, vocabulary, high=None, low=None,
```

Comment:
the name and argument of the function is not clear without documentation.
at least, `high` and `low` is strange to use, maybe `lower_boundary` and `upper_boundary` is better?

Reply:
`high`, `low` are fine when you see "limit" in the method's name. A bigger problem is the order of these two things, where naturally `low` should on the left side and `high` on the other side. Otherwise, if these two arguments are passed positionally instead of as keywords, it would be confusing.

Reply:
wait, so `high` and `low` are term frequencies? Now it's confusing...